t/33-vagrant.t

```diff
@@ -0,0 +1,524 @@
+#!/usr/bin/perl
+
+use Cwd qw(abs_path);
+use Test::Most;
+use File::chdir;
+use FindBin '$Bin';
+use lib "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+use Test::MockRandom 'backend::vagrant';
+use Test::MockObject;
+use Test::MockModule;
+use Test::Warnings ':report_warnings';
+use Test::Output qw(stderr_like);
+use backend::vagrant;
+
+use bmwqemu ();
+
+# hide debug messages, keep warnings & errors
+$bmwqemu::logger = Mojo::Log->new(level => 'warn');
+
+my $mock_vagrant = Test::MockModule->new("vagrant", no_auto => 1);
+$mock_vagrant->noop('mkdir');
+
+my $tmpdir        = File::Temp->newdir();
+my $mock_file_tmp = Test::MockModule->new("File::Temp");
+$mock_file_tmp->redefine('newdir', $tmpdir);
+
+my $expected_console_name        = undef;
+my $expected_console_type        = undef;
+my %expected_console_credentials = ();
+my $distri                       = Test::MockModule->new('distribution');
+$distri->redefine('add_console', sub {
+        my ($self, $name, $type, $creds) = @_;
+
+        local $Test::Builder::Level = $Test::Builder::Level + 1;
+
+        if (defined($expected_console_name)) {
+            die "Invalid console name '$name'. Expected '$expected_console_name'" unless $name eq $expected_console_name;
+        }
+        if (defined($expected_console_type)) {
+            die "Invalid console type '$type'. Expected '$expected_console_type'" unless $type eq $expected_console_type;
+        }
+        if (%expected_console_credentials) {
+            is_deeply($creds, \%expected_console_credentials, 'Check the name of the console');
+        }
+        my $ret = Test::MockObject->new();
+        $ret->set_true('backend');
+        return $ret;
+});
+$testapi::distri = distribution->new;
+
+my $check_socket_ret            = undef;
+my $check_socket_fh_expected    = undef;
+my $check_socket_write_expected = undef;
+my $mock_base                   = Test::MockModule->new("backend::baseclass", no_auto => 1);
+$mock_base->redefine('check_socket', sub {
+        my ($self, $fh, $write) = @_;
+        if (defined($check_socket_fh_expected)) {
+            die "Invalid fh '$fh'. Expected '$fh'" unless $fh eq $check_socket_fh_expected;
+        }
+        if (defined($check_socket_write_expected)) {
+            die "Invalid write '$write'. Expected '$check_socket_write_expected'" unless $write eq $check_socket_write_expected;
+        }
+        return $check_socket_ret;
+});
+
+
+my $mock_run  = Test::MockModule->new("IPC::Run");
+my $mock_file = Test::MockModule->new("File::chdir");
+
+my $expected_spurt_str = undef;
+my $mock_path          = Test::MockObject->new();
+$mock_path->mock('make_path', sub {
+        my ($self, %opts) = @_;
+        return $self;
+});
+$mock_path->mock('child', sub {
```

Comment:
With signatures code like this would possibly fit into one line.

Reply:
I've replaced all the old style `sub`s with signatures also in `33-vagrant.t`.